agenthub/codeact_swe_agent/codeact_swe_agent.py

```diff
@@ -52,10 +52,9 @@ def get_action_message(action: Action) -> dict[str, str] | None:
     return None
 
 
-def get_observation_message(obs) -> dict[str, str] | None:
-    max_message_chars = config.get_llm_config_from_agent(
-        'CodeActSWEAgent'
-    ).max_message_chars
+def get_observation_message(
+    obs: Observation, max_message_chars: int
```

Comment:
Optionally: make `get_observation_message()` a class method and it can directly use `self.llm.config.max_message_chars` , thus avoid extra code below to pass it as parameter.